lib/ClangImporter/ClangAdapter.h

```diff
@@ -60,6 +60,16 @@ struct PlatformAvailability;
 Optional<const clang::Decl *>
 getDefinitionForClangTypeDecl(const clang::Decl *D);
 
+/// Returns the first redeclaration of \p D outside of a function.
+///
+/// If there are no such redeclarations, returns null.
+/// If \p D is not redeclarable, just returns \p D itself.
+///
+/// Use this similarly to clang::Decl::getCanonicalDecl when the containing
+/// module matters.
```

Comment:
In what situations does the containing module not matter? Is this supposed to be clear from the usage context?

Reply:
:-/ I don't have an answer to this one. I looked at each of the call sites for `getCanonicalDecl` and determined that only one made sense to change because all the others weren't going to ask the question "which module are you from", but still.